src/wallet/rpc/spend.cpp

```diff
@@ -1457,6 +1457,102 @@ RPCHelpMan sendall()
+        "EXPERIMENTAL warning: this call may be changed in future releases.\n"
+        "\nSubmit a raw transaction to the wallet directly, possibly providing initial metadata.\n",
+        {
+            {"hexstring", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The hex string of the raw transaction"},
+            {"options", RPCArg::Type::OBJ_USER_KEYS, RPCArg::Optional::OMITTED_NAMED_ARG, "Options",
+                {
+                    {"foreign_outputs", RPCArg::Type::ARR, RPCArg::Default{UniValue::VARR}, "List of outputs sent by other wallets (eg, for a CoinJoin transaction)",
+                        {
+                            {"output_index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "Index of an output to mark as foreign"},
+                        },
+                    },
+                },
+            },
+        },
+        RPCResult{RPCResult::Type::NONE, "", ""},
+        RPCExamples{
+            "\nCreate a transaction\n"
+            + HelpExampleCli("createrawtransaction", "\"[{\\\"txid\\\" : \\\"mytxid\\\",\\\"vout\\\":0}]\" \"{\\\"myaddress\\\":0.01}\"") +
+            "Sign the transaction, and get back the hex\n"
+            + HelpExampleCli("signrawtransactionwithwallet", "\"myhex\"") +
+            "\nSend the transaction (signed hex)\n"
+            + HelpExampleCli("submitrawtransactiontowallet", "\"signedhex\"") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("submitrawtransactiontowallet", "\"signedhex\"")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            std::shared_ptr<CWallet> const pwallet{GetWalletForJSONRPCRequest(request)};
+            if (!pwallet) return UniValue::VNULL;
+
+            RPCTypeCheck(request.params, {
+                UniValue::VSTR, // hexstring
+                UniValue::VOBJ, // options
+            }, /*fAllowNull=*/true);
+
+            CMutableTransaction mtx;
+            if (!DecodeHexTx(mtx, request.params[0].get_str())) {
+                throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed. Make sure the tx has at least one input.");
+            }
+            CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
+
+            std::optional<std::vector<bool>> foreign_outputs;
+            if (!request.params[1].isNull()) {
+                const UniValue& options = request.params[1];
+                RPCTypeCheckObj(options, {
+                    {"foreign_outputs", UniValueType(UniValue::VARR)},
+                }, /*fAllowNull=*/true, /*fStrict=*/true);
+
+                if (!options["foreign_outputs"].isNull()) {
+                    auto output_count = tx->vout.size();
+                    foreign_outputs.emplace(output_count);
+                    const auto& foreign_outputs_uv = options["foreign_outputs"].get_array();
+                    size_t final_foreign_outputs_size = 0;
+                    for (auto i = foreign_outputs_uv.size(); i--; ) {
```

Comment:
node0 stderr wallet/rpc/spend.cpp:1516:63: runtime error: unsigned integer overflow: 0 - 1 cannot be represented in type 'size_t' (aka 'unsigned long')

Reply:
AFAIK this is defined behaviour and correct.

Reply:
This one didn't need the index at all, so changed it.

Opened https://github.com/llvm/llvm-project/issues/57794 to address the false positives directly, and added exceptions for the others for now.